module/metrics/transaction.go

```diff
@@ -81,11 +84,65 @@ func NewTransactionCollector(transactionTimings mempool.TransactionTimings, log
 			Subsystem: subsystemTransactionSubmission,
 			Help:      "counter for the success/failure of transaction submissions",
 		}, []string{"result"}),
+		executeScriptDuration: promauto.NewCounterVec(prometheus.CounterOpts{
+			Name:      "execute_script_rtt_duration",
+			Namespace: namespaceAccess,
+			Subsystem: subsystemTransactionSubmission,
+			Help:      "counter for the duration in ms of the round trip time for executing a script",
+		}, []string{"script_size"}),
+		transactionResultDuration: promauto.NewCounterVec(prometheus.CounterOpts{
+			Name:      "transaction_result_rtt_duration",
+			Namespace: namespaceAccess,
+			Subsystem: subsystemTransactionSubmission,
+			Help:      "counter for the duration in ms of the round trip time for getting a transaction result",
+		}, []string{"script_size"}),
+		scriptSize: promauto.NewHistogramVec(prometheus.HistogramOpts{
+			Name:      "script_size",
+			Namespace: namespaceAccess,
+			Subsystem: subsystemTransactionSubmission,
+			Help:      "histogram for the script size in kb histogram in GetTransactionResult and ExecuteScript",
+		}, []string{"method"}),
 	}
 
 	return tc
 }
 
+func (tc *TransactionCollector) ExecuteScriptRTT(dur time.Duration, size int) {
+	tc.scriptSizeHist(size, "ExecuteScript")
+	tc.executeScriptDuration.With(prometheus.Labels{
+		"script_size": tc.scriptSizeLabel(size),
+	}).Add(float64(dur.Milliseconds()))
+}
+
+func (tc *TransactionCollector) GetTransactionResultRTT(dur time.Duration, size int) {
+	tc.scriptSizeHist(size, "GetTransactionResult")
+	tc.transactionResultDuration.With(prometheus.Labels{
+		"script_size": tc.scriptSizeLabel(size),
+	}).Add(float64(dur.Milliseconds()))
+}
+
+func (tc *TransactionCollector) scriptSizeLabel(size int) string {
```

Comment:
Is there a way to get more accurate size labels ? Maybe only use the kb conversion for very large scripts.

Reply:
Currently using Vishal's recommendation https://github.com/dapperlabs/flow-internal/issues/1536#issuecomment-1122955228